metals/src/main/scala/scala/meta/internal/troubleshoot/ProblemResolver.scala

```diff
@@ -0,0 +1,161 @@
+    var deprecatedSbt = false
+    var futureSbt = false
+
+    for {
+      target <- allTargets
+      issue <- findProblem(target)
+    } yield {
+      issue match {
+        case UnsupportedScalaVersion(version) => unsupportedVersions += version
+        case DeprecatedScalaVersion(version) => deprecatedVersions += version
+        case FutureScalaVersion(version) => futureVersions += version
+        case _: SemanticDBDisabled => misconfiguredProjects += 1
+        case _: MissingSourceRoot => misconfiguredProjects += 1
+        case UnsupportedSbtVersion => unsupportedSbt = true
+        case DeprecatedSbtVersion => deprecatedSbt = true
+        case FutureSbtVersion => futureSbt = true
+      }
+    }
+
+    val unsupportedMessage = if (unsupportedVersions.nonEmpty) {
+      Some(Messages.UnsupportedScalaVersion.message(unsupportedVersions.toSet))
+    } else {
+      None
+    }
+    val deprecatedMessage = if (deprecatedVersions.nonEmpty) {
+      Some(Messages.DeprecatedScalaVersion.message(deprecatedVersions.toSet))
+    } else {
+      None
+    }
+
+    val futureMessage = if (futureVersions.nonEmpty) {
+      Some(Messages.FutureScalaVersion.message(futureVersions.toSet))
+    } else {
+      None
+    }
+
+    val deprecatedSbtMessage =
+      if (deprecatedSbt) Some(Messages.DeprecatedSbtVersion.message) else None
+    val unsupportedSbtMessage =
+      if (deprecatedSbt) Some(Messages.UnsupportedSbtVersion.message) else None
+    val futureSbtMessage =
+      if (deprecatedSbt) Some(Messages.FutureSbtVersion.message) else None
+
+    val semanticdbMessage =
+      if (
+        misconfiguredProjects == allTargets.size && misconfiguredProjects > 0
+      ) {
+        Some(Messages.CheckDoctor.allProjectsMisconfigured)
+      } else if (misconfiguredProjects == 1) {
+        val name = allTargets
+          .find(t => !t.isSemanticdbEnabled)
+          .map(_.displayName)
+          .getOrElse("<none>")
+        Some(Messages.CheckDoctor.singleMisconfiguredProject(name))
+      } else if (misconfiguredProjects > 0) {
+        Some(
+          Messages.CheckDoctor.multipleMisconfiguredProjects(
+            misconfiguredProjects
+          )
+        )
+      } else {
+        None
+      }
+
+    val allMessages = List(
+      deprecatedMessage,
+      unsupportedMessage,
+      futureMessage,
+      deprecatedSbtMessage,
+      unsupportedSbtMessage,
+      futureSbtMessage,
+      semanticdbMessage
+    ).flatten
+
+    def scalaVersionsMessages = List(
+      deprecatedMessage,
+      unsupportedMessage,
+      futureMessage
+    ).flatten
+
+    allMessages match {
+      case single :: Nil => Some(single)
+      case Nil => None
+      case messages if messages == scalaVersionsMessages =>
+        Some(
+          s"Your build definition contains multiple unsupported and deprecated Scala versions."
+        )
+      case _ =>
+        Some(
+          s"Multiple problems detected in your build."
+        )
```

Comment:
So in this scenario, we do all the work up above to potentially get multiple messages, but then we never show them to the user, we just show `Multiple problems detected in your build`? Wouldn't it be better to actually show them to the user? I'd assume that `problemSummary` would include the summary of issues, not just a message saying there is issues.

Reply:
So problem summary will be shown as a message to the user and if we get too much problems then it will be quite bloated. 

`Multiple problems detected in your build` will be accompanied by `More information`, so people can easily take a look. Though I have never honestly seen more than one issue at the same time, so it's highly improbable scenario.

Reply:
Ah ok I totally missed the `More information` part. If that's there, then nevermind.